app/controllers/users.js

```diff
@@ -0,0 +1,52 @@
+'use strict';
+
+const userService = require('../services/users'),
+  errors = require('../errors');
+
+const checkAlphanumeric = (string) => {
+  const regex = /^[a-zA-Z0-9]*$/;
+  return regex.test(string);
+};
+
+const validateEmail = (email) => {
+  const regex = /\S+@\S+\.\S+/;
+  return regex.test(email);
+};
+
+const emailValidation = (email) => {
+  if (!validateEmail(email)) {
+    return 'The requested email is not a valid email';
+  } else if (email.indexOf('@wolox.com.ar') < 0) {
+    return 'The requested email is not a Wolox email';
+  } else {
+    return undefined;
+  }
+};
+
+const passwordValidation = (password) => {
+  if (!checkAlphanumeric(password)) {
+    return 'The password must be alphanumeric';
+  } else if (password.length < 8) {
+    return 'The password is too short';
+  } else {
+    return undefined;
+  }
+};
+
+exports.signup = (req, res, next) => {
+  const saltRounds = 10;
+  const newUser = req.body;
+  let validationErrors = {};
+  validationErrors = {
+    email: emailValidation(newUser.email),
+    password: passwordValidation(newUser.password)
+  };
+  if (validationErrors.email || validationErrors.password) {
+    next(validationErrors.email ? errors.defaultError(validationErrors.email) : errors.defaultError(validationErrors.password));
```

Comment:
We shouldn't be responding with a defaultError because that would lead to a 500 status code, and in these cases we will want some status code more like 4xx.